File: PB002.fs/Algorithm.fs
namespace ProjectEuler

type public Algorithm() = class
    let fibonacci = 
        let rec iter a b =
            seq {
                yield a
                yield! iter b (a + b)
            }
        iter 1 2

    let run = 
        fibonacci 
        |> Seq.takeWhile (fun (x) -> (x <= 4000000))
        |> Seq.filter (fun (x) -> ((x &&& 1) = 0))
        |> Seq.sum
 
    interface IAlgorithm with
        member this.Compute(): string = run.ToString()
        member this.Prepare(): bool = true
    end
end

File: PB009.cs/Algorithm.cs
using System;

namespace ProjectEuler
{
    public class Algorithm : IAlgorithm
    {
        public string Compute()
        {
            int a = 0, b = 0, c = 0;
            for (a = 1; a < 1000; a++)
                for (b = a; (b < 1000) && (a + b < 1000); b++)
                {
                    c = 1000 - a - b;
                    if (a * a + b * b == c * c)
                        return (a * b * c).ToString();
                }
            return "";
        }

        public bool Prepare()
        {
            return true;
        }
    }
}
